manager/controllers/app/plotter_controller.go

```diff
@@ -0,0 +1,246 @@
+			// the finalizer is present - delete the allocated resources
+
+			for cluster, blueprint := range plotter.Status.Blueprints {
+				// TODO Check namespace deletion. Some finalizers leave namespaces in terminating state
+				err := r.ClusterManager.DeleteBlueprint(cluster, blueprint.Namespace, blueprint.Name)
+				if err != nil {
+					return err
+				}
+			}
+
+			// remove the finalizer from the list and update it, because it needs to be deleted together with the object
+			ctrlutil.RemoveFinalizer(plotter, finalizerName)
+
+			if err := r.Update(context.Background(), plotter); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+	// Make sure this CRD instance has a finalizer
+	if !hasFinalizer {
+		ctrlutil.AddFinalizer(plotter, finalizerName)
+		if err := r.Update(context.Background(), plotter); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *PlotterReconciler) reconcile(plotter *app.Plotter) (ctrl.Result, error) {
+	isInitialReconcile := false
+	if plotter.Status.Blueprints == nil {
+		plotter.Status.Blueprints = make(map[string]app.MetaBlueprint)
+		isInitialReconcile = true
+	}
+
+	// Reconciliation loop per cluster
+	isReady := true
+	for cluster, blueprintSpec := range plotter.Spec.Blueprints {
+		r.Log.V(1).Info("Handling cluster " + cluster)
+		if blueprint, exists := plotter.Status.Blueprints[cluster]; exists {
+			r.Log.V(2).Info("Found status for cluster " + cluster)
+
+			remoteBlueprint, err := r.ClusterManager.GetBlueprint(cluster, blueprint.Namespace, blueprint.Name)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+
+			if remoteBlueprint == nil {
+				r.Log.Info("Could not yet find remote blueprint")
+				return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
+			}
+
+			if !reflect.DeepEqual(blueprintSpec, remoteBlueprint.Spec) {
+				r.Log.V(1).Info("Blueprint specs differ...")
+				remoteBlueprint.Spec = blueprintSpec
+				remoteBlueprint.ObjectMeta.Annotations = map[string]string(nil) // reset annotations
+				err := r.ClusterManager.UpdateBlueprint(cluster, remoteBlueprint)
+				if err != nil {
+					return ctrl.Result{}, err
+				}
+				return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
+			}
+
+			r.Log.V(2).Info("Status of remote blueprint ", "status", remoteBlueprint.Status)
+
+			blueprint.Status = remoteBlueprint.Status
+
+			plotter.Status.Blueprints[cluster] = blueprint
+
+			if !remoteBlueprint.Status.ObservedState.Ready {
+				isReady = false
+			}
+
+			// If Blueprint has an error set it as status of plotter
+			if remoteBlueprint.Status.ObservedState.Error != "" {
+				plotter.Status.ObservedState.Error = remoteBlueprint.Status.ObservedState.Error
+			}
+		} else {
+			random := rand.String(5)
+			randomNamespace := "m4d-" + random
+			blueprint := &app.Blueprint{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Blueprint",
+					APIVersion: "app.m4d.ibm.com/v1alpha1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "blueprint",
```

Comment:
Do we want to match the plotter name, for an easier debug (in case more than one plotter is active)?